vscode-dotnet-runtime-extension/src/extension.ts

```diff
@@ -432,6 +443,95 @@ export function activate(vsCodeContext: vscode.ExtensionContext, extensionContex
         return uninstall(commandContext);
     });
 
+    /**
+     * @param commandContext The context of the request to find the dotnet path.
+     * We wrap an AcquisitionContext which must include the version, requestingExtensionId, architecture of .NET desired, and mode.
+     * The architecture should be of the node format ('x64', 'x86', 'arm64', etc.)
+     *
+     * @returns the path to the dotnet executable, if one can be found. This should be the true path to the executable. undefined if none can be found.
+     *
+     * @remarks Priority Order for path lookup:
+     * VSCode Setting -> PATH -> Realpath of PATH -> DOTNET_ROOT (Emulation DOTNET_ROOT if set first)
+     *
+     * This accounts for pmc installs, snap installs, bash configurations, and other non-standard installations such as homebrew.
+     */
+    const dotnetFindPathRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.findPath}`, async (commandContext : IDotnetFindPathContext) =>
+    {
+        globalEventStream.post(new DotnetFindPathCommandInvoked(`The find path command was invoked.`, commandContext));
+
+        if(!commandContext.acquireContext.mode || !commandContext.acquireContext.requestingExtensionId || !commandContext.acquireContext.version || !commandContext.acquireContext.architecture)
+        {
+            throw new EventCancellationError('BadContextualFindPathError', `The find path request was missing required information: a mode, version, architecture, and requestingExtensionId.`);
+        }
+
+        globalEventStream.post(new DotnetFindPathLookupSetting(`Looking up vscode setting.`));
+        const workerContext = getAcquisitionWorkerContext(commandContext.acquireContext.mode, commandContext.acquireContext);
+        const existingPath = await resolveExistingPathIfExists(existingPathConfigWorker, commandContext.acquireContext, workerContext, utilContext, commandContext.versionSpecRequirement);
+
+        if(existingPath)
+        {
+            globalEventStream.post(new DotnetFindPathSettingFound(`Found vscode setting.`));

```

Comment:
Q: If this is coming from a setting, is this just an abitrary value. Do you need to verify that the path actually exists or is that responsibility deferred to something else?

Reply:
Good question, we do verify the path in the function/classes that get the path setting. That was added in the last version.